Fixing magellan is-active issue  ||  adding traversing to magelan $active

h-e-r-b-e-r-t opened the pull request:
[https://github.com/zurb/foundation-sites/issues/10470#issuecomment-319499782](https://github.com/zurb/foundation-sites/issues/10470#issuecomment-319499782)

__menu.scss_ &&  __dropdown-menu.scss_:
Fixing: '_**is-active**_' class is added to a-tag but all scss is for parent element to receive  is-active class.
Making _**is-active**_ class general => see changes in magellan.js

_magellan.js_:
Adding option _**activeParents**_ for traversing $active element.

h-e-r-b-e-r-t committed:
magellan: adding flexibility to $active

/* https://github.com/zurb/foundation-sites/issues/10470#issuecomment-319499782 */
Adding option activeParents for traverse targeting $active.

h-e-r-b-e-r-t committed:
Update _menu.scss: making '.is-active' general

https://github.com/zurb/foundation-sites/issues/10470#issuecomment-319499782
Fixing: 'is-active' class is added to a-tag but all scss is for parent element to receive  is-active class.
Making 'is-active' class general => see changes in magellan.js

h-e-r-b-e-r-t committed:
_dropdown-menu.scss: making '.is-active' general

https://github.com/zurb/foundation-sites/issues/10470#issuecomment-319499782
Fixing: 'is-active' class is added to a-tag but all scss is for parent element to receive  is-active class.
Making is-active class general => see changes in magellan.js

SassNinja commented on js/foundation.magellan.js:
```diff
@@ -165,8 +165,10 @@ class Magellan extends Plugin {
+  	               ? $( this.$links.filter('[href="#' + this.$targets.eq(curIdx).data('magellan-target') + '"]').parents()[this.options.activeParents] ).addClass(this.options.activeClass)
+  	               : this.$links.filter('[href="#' + this.$targets.eq(curIdx).data('magellan-target') + '"]').addClass(this.options.activeClass);
+    
     if(this.options.deepLinking){
```
You shouldn't travel up the complete DOM tree twice!
I would rather write it this way and use jQuery's `eq()` so you don't have to recreate a jQuery object.

```javascript
this.$active = this.options.activeParents
                ? this.$links.filter('[href="#' + this.$targets.eq(curIdx).data('magellan-target') + '"]').parents().eq(this.options.activeParents).addClass(this.options.activeClass)
                : this.$links.filter('[href="#' + this.$targets.eq(curIdx).data('magellan-target') + '"]').addClass(this.options.activeClass);
```

SassNinja commented on js/foundation.magellan.js:
```diff
@@ -238,6 +240,13 @@ Magellan.defaults = {
+  * @default -1
+  */
+ activeParents: -1,
   /**
```
If you use jQuery's `eq()` as suggested above, this has to be `null`default and e.g. `0` if you want the first parent.

SassNinja commented on scss/components/_dropdown-menu.scss:
```diff
@@ -152,7 +152,7 @@ $dropdown-menu-item-background-active: transparent !default;
     // Active state
-    .is-active > a {
+    .is-active {
       background: $dropdown-menu-item-background-active;
```
This breaks existing projects that rely on `is-active` to be on the links's parent instead of on the link itself.
Thus I wouldn't change this or extend it:
```sass
.is-active {
    &, > a {
```

SassNinja commented:
@h-e-r-b-e-r-t I've reviewed your code.

Apart from my comments I'm wondering if we really need an option travel up the DOM multiple levels?
I'd prefer to only offer one level, so rather a boolean (activeParent) than a number what lets you replace `parents().eq()` with `parent()`.

@kball @IamManchanda what do you think?

h-e-r-b-e-r-t committed:
Update _dropdown-menu.scss

kball commented on scss/components/_menu.scss:
```diff
@@ -441,7 +441,7 @@ $menu-icons-back-compat: true !default;
 
     // Active state
-    .is-active > a {
+    .is-active {
```
This seems like a pretty big change... what is the thinking here?

h-e-r-b-e-r-t committed:
Multiple magellan on one page

https://foundation.zurb.com/forum/posts/37575-multiple-magellan-on-one-page?sort=Most+Helpful

DanielRuf commented:
What is the current state of this PR?

kball commented:
Closing this PR as the feedback was never addressed or responded to.
